diesel/src/pg/expression/expression_methods.rs

```diff
@@ -211,3 +211,93 @@ impl<T, ST> ArrayExpressionMethods<ST> for T where
     T: Expression<SqlType=Array<ST>>,
 {
 }
+
+use expression::predicates::{Asc, Desc};
+
+pub trait SortExpressionMethods : Sized {
+    /// Specify that nulls should come before other values in this ordering.
+    /// Normally, nulls come last when sorting in ascending order and first
+    /// when sorting in descending order.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # #[macro_use] extern crate diesel;
+    /// # include!("src/doctest_setup.rs");
+    /// #
+    /// # table! {
+    /// #     users {
+    /// #         id -> Integer,
+    /// #         name -> VarChar,
+    /// #     }
+    /// # }
+    /// #
+    /// # table! {
+    /// #     foos {
+    /// #         id -> Integer,
+    /// #         foo -> Nullable<Integer>,
+    /// #     }
+    /// # }
+    /// #
+    /// # fn main() {
+    /// #     let connection = connection_no_data();
+    /// #     connection.execute("drop table if exists foos").unwrap();
+    /// #     connection.execute("create table foos ( id SERIAL PRIMARY KEY, foo Integer )").unwrap();
+    /// #     connection.execute("insert into foos (foo) values (null), (1), (2)").unwrap();
+    /// #     use self::foos::dsl::*;
+    /// assert_eq!(Ok(vec!(Some(1), Some(2), None)),
+    ///            foos.select(foo).order(foo.asc()).load(&connection));
```

Comment:
The thing I really want to show is that the order of the elements, in this case `vec![Some(3), Some(17), Some(4711), None, None]` would have been equally correct, but `vec![Some(2), None, Some(1)]` would have been obviously wrong.  Right?

But actually showing the table definition and contents might be a good thing.  I'll take a look at it.

The bracket thing: Yes, will fix, thank you!